qiskit/wrapper/credentials/_configrc.py

```diff
@@ -0,0 +1,138 @@
+
+    Args:
+        filename (str): full path to the qiskitrc file. If `None`, the default
+            location is used (`HOME/.qiskit/qiskitrc`).
+
+    Returns:
+        dict: dictionary with the contents of the configuration file, with
+            the form::
+
+            {'provider_class_name': {'token': 'TOKEN', 'url': 'URL', ... }}
+
+    Raises:
+        QISKitError: if the file was not parseable. Please note that this
+            exception is not raised if the file does not exist (instead, an
+            empty dict is returned).
+    """
+    filename = filename or DEFAULT_QISKITRC_FILE
+    config_parser = ConfigParser()
+    try:
+        config_parser.read(filename)
+    except ParsingError as ex:
+        raise QISKitError(str(ex))
+
+    # Build the credentials dictionary.
+    credentials_dict = {}
+    for name in config_parser.sections():
+        single_credentials = dict(config_parser.items(name))
+        # TODO: 'proxies' is the only value that is a dict. Consider moving to
+        # json configuration or splitting into single keys manually.
+        if 'proxies' in single_credentials.keys():
+            single_credentials['proxies'] = literal_eval(
+                single_credentials['proxies'])
+        credentials_dict[name] = single_credentials
+
+    return credentials_dict
+
+
+def write_qiskit_rc(credentials, filename=None):
+    """
+    Write credentials to the configuration file.
+
+    Args:
+        credentials (dict): dictionary with the credentials, with the form::
+            {'provider_class_name': {'token': 'TOKEN', 'url': 'URL', ... }}
+        filename (str): full path to the qiskitrc file. If `None`, the default
+            location is used (`HOME/.qiskit/qiskitrc`).
+    """
+    filename = filename or DEFAULT_QISKITRC_FILE
+    # Create the directories and the file if not found.
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+
+    # Write the configuration file.
+    with open(filename, 'w') as config_file:
+        config_parser = ConfigParser()
+        config_parser.read_dict(credentials)
+        config_parser.write(config_file)
+
+
+def store_credentials(provider_class=IBMQProvider, overwrite=False,
+                      filename=None, **kwargs):
+    """
+    Store the credentials for a single provider in the configuration file.
+
+    Args:
+        provider_class (class): class of the Provider for the credentials.
+        overwrite (bool): overwrite existing credentials.
+        filename (str): full path to the qiskitrc file. If `None`, the default
+            location is used (`HOME/.qiskit/qiskitrc`).
+        kwargs (dict): keyword arguments passed to provider class
+            initialization.
+
+    Raises:
+        QISKitError: If provider already exists and overwrite=False; or if
+            the account_name could not be assigned.
+    """
+    # Set the name of the Provider from the class.
+    account_name = provider_class.__name__
```

Comment:
Can we use `"{}.{}".format(provider_class.__module__, provider_class.__name__)`? This way sure it is unique.